src/ll.GoogleTranslator.js

```diff
@@ -0,0 +1,131 @@
+/*!
+ * LL two-way parallel translation - Google translator class
+ *
+ * @copyright 2019 LL team and others; see LICENSE.txt for terms
+ */
+
+/**
+ * Google translator
+ * @abstract
+ * @class
+ *
+ * @constructor
+ * @param {Object} config
+ * @param {string} config.key API key
+ */
+ll.GoogleTranslator = function LLGoogleTranslator( config ) {
+	this.url = 'https://translation.googleapis.com';
+	if ( !config.key ) {
+		throw new Error( 'Need Google API key' );
+	}
+	this.key = config.key;
+	// Parent constructor: must be called after setting .url and .key above
+	ll.GoogleTranslator.super.call( this );
+};
+
+/* Initialization */
+
+OO.inheritClass( ll.GoogleTranslator, ll.Translator );
+
+// TODO: Make separators more robust.
+// Google messes with multi-character separators, usually by adding spaces betweem unalike character
+// repetitions, such as ":!!:" becoming ": !!:". Using more obscure characters works, but is still
+// inherently fragile.
+ll.GoogleTranslator.static.outerSeparator = '☆';
+ll.GoogleTranslator.static.innerSeparator = '️★';
+
+/**
+ * List of queued tasks.
+ * @static
+ * @type {Array}
+ */
+ll.GoogleTranslator.static.stack = [];
+
+/**
+ * Interval timer for processing the queue.
+ * @static
+ * @type {Array}
+ */
+ll.GoogleTranslator.static.timer = null;
+
+/**
+ * Process queued tasks.
+ *
+ * Clears the processing interval when the queue is empty.
+ */
+ll.GoogleTranslator.static.process = function () {
+	var task = this.stack.shift();
+	if ( this.stack.length === 0 ) {
```

Comment:
This means that if this.stack contains more than one thing, this function will take the first thing off the front, throw it away, and do nothing, leaving the associated promise (returned by .queue()) pending forever. Is that intended? Do you mean for only the last-queued task to be executed, and all the others to be discarded? If that's the case, there are other ways of doing it that would make it clearer what's happening.

Reply:
Yeah, we throw them away because they aren't needed.